src/sdam/server.ts

```diff
@@ -500,7 +500,8 @@ function makeOperationHandler(
 ): Callback {
   const session = options?.session;
   return function handleOperationResult(error, result) {
-    if (result != null) {
+    // We should not swallow an error if it is present.
+    if (error == null && result != null) {
```

Comment:
If we are not montoring commands the callback doesn't get wrapped and thus both an error and result are present here.